src/arrayMethodSort.js

```diff
@@ -4,8 +4,29 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    for (let i = 0; i < this.length; i++) {
+      for (let k = 0; k < this.length; k++) {
+        const comparedEl = this[i];
+        const currentEl = this[k];
+
+        if (compareFunction === undefined) {
+          if (this[i].toString() < this[k].toString()) {
+            this[i] = currentEl;
+            this[k] = comparedEl;
+          }
+
+          continue;
+        }
+
+        if (compareFunction(this[i], this[k]) < 0) {
+          this[i] = currentEl;
+          this[k] = comparedEl;
+        }
```

Comment:
When a custom `compareFunction` is provided, it should be used to determine the sort order. If `compareFunction(a, b)` returns a value less than 0, it means that `a` comes before `b`. However, you are comparing `this[i]` with `this[k]` which may not be adjacent elements due to the incorrect loop setup. Ensure the elements being compared are adjacent after fixing the inner loop index.